infra/image/system-service/fixnet.sh

```diff
@@ -1,24 +1,62 @@
 #!/bin/bash -eu
 
+function validFQDN()
+{
+    local name="${1}"
+
+    [[ "${name}" =~ [[:space:]] ]] && return 1
+    [[ "${name}" =~ \. ]] || return 1
+    [[ "${name}" =~ \.\. ]] && return 1
+    for i in ${name//./ }; do
+        [[ "${i}" =~ ^[a-z0-9_/]+$ ]] || return 1
+    done
+    [[ "${name}" == "localhost.localdomain" ]] && return 1
+    return 0
+}
+
+function validIPv4()
+{
+    local ip="${1}"
+    local rematch="^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$"
+
+    [[ "${ip}" =~ ${rematch} ]] || return 1
+    for i in ${ip//./ }; do
+        [[ ${i} -le 255 ]] || return 1
+    done
+
+    return 0
+}
+
 HOSTNAME=$(hostname)
 IP=$(hostname -I | cut -d " " -f 1)
 
-if [ -z "${HOSTNAME}" ]; then
+if [ -z "${HOSTNAME}" ] || ! validFQDN "${HOSTNAME}" ; then
     echo "ERROR: Failed to retrieve hostname."
     exit 1
 fi
-if [ -z "${IP}" ]; then
-    echo "ERROR: Failed to retrieve IP address."
+if [ -z "${IP}" ] || ! validIPv4 "${IP}" ; then
+    echo "ERROR: Got invalid IPv4 address: '${IP}'"
     exit 1
 fi
 
-# shellcheck disable=SC2143
-if [ -n "$(grep -P "[[:space:]]${HOSTNAME}" /etc/hosts)" ]; then
+echo "Fix NET:"
+echo "  HOSTNAME: '${HOSTNAME}'"
+echo "  IP: '${IP}'"
+echo
+
+if grep -Pq "[[:space:]]${HOSTNAME}([[:space:]]|$)" /etc/hosts
+then
     sed -ie "s/.*${HOSTNAME}/${IP}\t${HOSTNAME}/" /etc/hosts
```

Comment:
sed sometimes is risky, 

what about 

```
sed -i.bak
```
it will create a backup before modifying.

Reply:
Fixed